src/core/exec-invoke.c

```diff
@@ -2692,7 +2692,7 @@ static int compile_bind_mounts(
 
                         /* When one of the parent directories is in the list, we cannot create the symlink
                          * for the child directory. See also the comments in setup_exec_directory(). */
-                        if (i->only_create)
+                        if (FLAGS_SET(i->flags, EXEC_DIRECTORY_ONLY_CREATE) && !FLAGS_SET(i->flags, EXEC_DIRECTORY_READ_ONLY))
```

Comment:
i don't grok the readonly flag conditionalization here?

Reply:
If the r/o flag is set, then we need to mount this, to make it r/o

Reply:
please add that as acomment

Reply:
not addressed